extensions/rich_text_components/Image/directives/ImageDirective.js

```diff
@@ -33,9 +34,16 @@ oppia.directive('oppiaNoninteractiveImage', [
       controller: ['$scope', '$attrs', function($scope, $attrs) {
         $scope.filepath = HtmlEscaperService.escapedJsonToObj(
           $attrs.filepathWithValue);
-        $scope.imageUrl = $sce.trustAsResourceUrl(
-          '/imagehandler/' + ExplorationContextService.getExplorationId() +
-          '/' + encodeURIComponent($scope.filepath));
+        ImagePreloaderService.addToRecentlyRequestedImageFilenames(
+          $scope.filepath);
+        $scope.imageUrl = '';
+        ImagePreloaderService.getImageUrl($scope.filepath)
+          .then(function(objectUrl) {
+          $scope.imageUrl = objectUrl; 
+          });
+        // [TODO] Display a loading indicator instead. For now, if the
```

Comment:
Missing implementation?

Reply:
Yes that will be the part of the next PR. Just added so that I know later on where to look at.